### PR TITLE
WT-8554 Automated testing for upgrade from 4.4 releases

### PR DESCRIPTION
**Pull request by AlexeyAnisimovAU**

### CODE REVIEW

File: test/evergreen/compatibility_test_for_releases.sh [hunk 1]
```diff
@@ -327,6 +339,52 @@ top="test-compatibility-run"
...
+    git clone --quiet --depth 1 --filter=blob:none --no-checkout https://github.com/wiredtiger/mongo-tests.git
+    cd mongo-tests; git checkout --quiet master -- WT-8395; cd WT-8395; test_data=$(pwd)
+
+    for FILE in $test_data/*; do tar -zxf $FILE; done
+    rm *.tar.gz; cd ../..
+
+    test_root=$(pwd)
+    for b in ${upgrade_to_latest_upgrade_downgrade_release_branches[@]}; do
+        (build_branch $b)
+        cd $b/test/checkpoint
```
Comment (alisonfel) [inline test/evergreen/compatibility_test_for_releases.sh:354]: Could we lift the contents of this for loop (L354-L383) into a function. More or less following the convention of the other tests in this file e.g. `run_format`/`verify_branches` (keeping this test modular).

File: test/evergreen/compatibility_test_for_releases.sh [hunk 2]
```diff
@@ -327,6 +339,52 @@ top="test-compatibility-run"
...
+            elif [[ "$test_res" != 0 ]]; then
+                echo "Error: Upgrade failed! Test result is $test_res."
+                exit 1
+            fi
+
+        done
+        cd $test_root
+    done
+
+    exit 0
```
Comment (alisonfel) [inline test/evergreen/compatibility_test_for_releases.sh:386]: I think we this exit should be removed. If we `exit 0` here, any other options passed to the script will be ignored i.e. `-n`,`-o`. I believe the script is supposed be able to run all the test options in a single execution.

File: test/evergreen/compatibility_test_for_releases.sh (hunk 1 above)
Comment (AlexeyAnisimovAU) [reply test/evergreen/compatibility_test_for_releases.sh]: Great point! Moved the code into a new `test_upgrade_to_branch` function.

File: test/evergreen/compatibility_test_for_releases.sh (hunk 2 above)
Comment (AlexeyAnisimovAU) [reply test/evergreen/compatibility_test_for_releases.sh]: Done!

File: test/evergreen/compatibility_test_for_releases.sh [hunk 3]
```diff
@@ -256,10 +256,53 @@ upgrade_downgrade()
...
+                    echo "Databases generated with unclean shutdown from versions 4.4.[0-6] must fail!"
+                    exit 1
+                fi
+            elif [[ "$test_res" != 0 ]]; then
+                echo "$output"
+                echo "Error: Upgrade of $FILE database to $1 failed! Test result is $test_res."
+                exit 1
+            fi
+        done
+        cd $3
```
Comment (kommiharibabu) [inline test/evergreen/compatibility_test_for_releases.sh:298]: I don't see a need of passing $3 just for cd operation, why can't we do it in the caller function?

File: test/evergreen/compatibility_test_for_releases.sh [hunk 4]
```diff
@@ -327,6 +381,21 @@ top="test-compatibility-run"
 rm -rf "$top" && mkdir "$top"
 cd "$top"
 
+if [ "$upgrade_to_latest" = true ]; then
+    # Prepare test data.
+    echo "Preparing test data..."
+    git clone --quiet --depth 1 --filter=blob:none --no-checkout https://github.com/wiredtiger/mongo-tests.git
+    cd mongo-tests; git checkout --quiet master -- WT-8395; cd WT-8395; test_data=$(pwd)
+
+    for FILE in $test_data/*; do tar -zxf $FILE; done
```
[the remark this answers is not in the document]
Comment (AlexeyAnisimovAU) [reply test/evergreen/compatibility_test_for_releases.sh:390]: Good catch, thank you!

**Comment by AlexeyAnisimovAU** [comment]
I have addressed all the comments. Please do another round of review.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/evergreen.yml [hunk 5]
```diff
@@ -4213,6 +4224,7 @@ buildvariants:
   - ubuntu2004-test
   tasks:
     - name: compatibility-test-for-newer-releases
+    - name: compatibility-test-upgrade-to-latest
```
Comment (lukech) [inline test/evergreen.yml:4227]: Putting the new task here will get it run for each commit build. Should we consider running it less frequently (by putting it into the `compatibility-tests-less-frequent` builder)?

**Review by lukech — APPROVED** [review]
Well done, Alexey! Just a minor comment about the scheduling frequency for the new task. Do you have a patch build to test the new task?

**Comment by AlexeyAnisimovAU** [comment]
> Well done, Alexey! Just a minor comment about the scheduling frequency for the new task. Do you have a patch build to test the new task?

Thanks @lukech! As discussed in Slack, I've added a new build variant "compatibility-tests-daily".

**Review by alisonfel — APPROVED** [review]
LGTM 👍

**Comment by lukech** [comment]
@AlexeyAnisimovAU LGTM. Thanks for making the change.

**Comment by AlexeyAnisimovAU** [comment]
Thank you everyone for the reviews!

**Pull request merged**